lib/chef-dk/policyfile/differ.rb

```diff
@@ -0,0 +1,264 @@
+#
+# Copyright:: Copyright (c) 2015 Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'diff/lcs'
+require 'paint'
+
+module ChefDK
+  module Policyfile
+    class Differ
+
+      POLICY_SECTIONS = %w{ revision_id run_list named_run_lists cookbook_locks default_attributes override_attributes }.freeze
+      LINES_OF_CONTEXT = 3
+      INITIAL_FILE_LENGTH_DIFFERENCE = 0
+      FORMAT = :unified
+
+
+      attr_reader :old_lock
+      attr_reader :old_name
+      attr_reader :new_lock
+      attr_reader :new_name
+      attr_reader :ui
+
+      def initialize(old_name: nil, old: nil, new_name: nil, new: nil, ui: nil)
```

Comment:
I think `old_lock` and `new_lock` are better than `old` and `new` because `new` is so commonly associated with class initialization